libbeat/outputs/elasticsearch/client.go

```diff
@@ -235,7 +237,7 @@ func (client *Client) publishEvents(ctx context.Context, data []publisher.Event)
 		failedEvents = data
 		stats.fails = len(failedEvents)
 	} else {
-		failedEvents, stats = bulkCollectPublishFails(client.log, result, data)
+		failedEvents, stats = bulkCollectPublishFails(client.log, result, data, client.NonIndexableAction)
```

Comment:
Nit: `buldCollectPublishFails` could be a function of `Client` so you would not have to pass the logger and the action to the function.

Reply:
Noticed this too, but was going with the 'when in Rome' approach. Will update.